linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/SourceMapWriter.scala

```diff
@@ -269,17 +269,30 @@ final class SourceMapWriter(out: ByteArrayWriter, jsFileName: String,
   protected def doWriteSegment(columnInGenerated: Int, pos: Position, name: String): Unit = {
     // scalastyle:off return
 
+    /* This method is incredibly performance-sensitive, so we resort to
+     * "unsafe" direct access to the underlying array of `out`.
```

Comment:
Have you compared this with allocating a buffer just for `doWriteSegment` (and then writing it out to the stream in a single `write`)?

I'm wondering if we're hitting some cache locality snafu here.

Reply:
I still have to test that variant.

Reply:
Well, that one is worse than the `ByteBuffer`. It's a 17% regression. :p

So I'll merge as is.